scripts/generate_ssl_debug_helpers.py

```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/env python3
+
+"""Generate library/ssl_debug_helps_generated.c
+
+The code generated by this module includes debug helper functions that can not be
+implemented by fixed codes.
+
+"""
+
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import sys
+import re
+import os
+import textwrap
+from mbedtls_dev import build_tree
+
+def remove_c_comments(string):
```

Comment:
Parsing C with regexes feels a bit brittle - maybe OK for this use case but using proper parsing tools might be nicer. I'm inclined to go with this rather than suggest we spend time re-writing using eg pycparser though.

Reply:
I have checked pycparser. I haven't got how to apply it. Maybe we can apply it in future.